app/models/piece.rb

```diff
@@ -8,14 +8,44 @@ def move_to!(params)
     new_x = params[:horizontal_position].to_i
     new_y = params[:vertical_position].to_i
     @chess_piece = game.pieces.find_by(horizontal_position: new_x, vertical_position: new_y)
-    if @chess_piece && valid_move?(new_x, new_y)
+    enemy_pawn = enemy_location(color, new_x, vertical_position)
+    if capture?(new_x, new_y)
       capture!(new_x, new_y)
+    elsif double_move?(type, vertical_position, new_x, new_y)
+      update_it!(new_x, new_y)
+      update_attributes(passant: true)
     elsif valid_move?(new_x, new_y)
-      update_attributes(horizontal_position: new_x, vertical_position: new_y, castle: false)
+      passant_capture!(enemy_pawn, type)
+      update_it!(new_x, new_y)
     elsif can_castle?(new_x, new_y, color)
       perform_castling(new_x, color)
     else
       false
     end
   end
+
+  def enemy_location(color, new_x, vertical_position)
+    if color == 'white'
+      game.pieces.find_by(type: 'Pawn', horizontal_position: new_x, vertical_position: vertical_position, color: 'black', passant: true)
+    else
+      game.pieces.find_by(type: 'Pawn', horizontal_position: new_x, vertical_position: vertical_position, color: 'white', passant: true)
+    end
+  end
```

Comment:
This can just be:

``` ruby
def enemy_location(color, new_x, vertical_position)
  game.pieces.find_by(type: 'Pawn', horizontal_position: new_x, vertical_position: vertical_position, color: color, passant: true)
end
```